src/routes/targeted-messaging/targeted-messaging.service.ts

```diff
@@ -44,27 +45,54 @@ export class TargetedMessagingService {
     }
   }
 
+  private async getOrCreateTargetedSafe(args: {
```

Comment:
If the survey is public and not targeted, why do we need to call `createTargetedSafes`? Is it possible to simply create a submission once the user has submitted the survey? Creating a separate entity for each Safe seems unnecessary and could overwhelm the database.

Reply:
Just because each submission requires a link to an entry in targeted_safes. We could maybe get rid of this requirement and have an option to store the safe address directly in each submission instead?

Reply:
I see. Since it’s not targeted, I find it confusing to create a new entity in the targeted safe table. Additionally, the name of the method, `createTargetedSafe`, is misleading for me. At first glance, it gives the impression that the method creates a new safe account. What do you think?

Reply:
renamed `createTargetedSafe` to  `addSafeToOutreach`: https://github.com/safe-global/safe-client-gateway/pull/2194/commits/7f93ed57b5881c3474a6fb2f6d0550d583647041

If you're ok to leave the creation of new entries in `targeted_safes` for now like we discussed on slack, we can handle this in the next iteration.

Reply:
Sure, Let's do that in another PR.